src/backend/cdb/cdbdtxrecovery.c

```diff
@@ -558,17 +561,38 @@ redoDistributedCommitRecord(TMGXACT_LOG *gxact_log)
 
 	if (i == *shmNumCommittedGxacts)
 	{
+		gxacts_limit = max_tm_gxacts;
+
+#ifdef FAULT_INJECTOR
+		if (SIMPLE_FAULT_INJECTOR("gxacts_overflow") == FaultInjectorTypeSkip)
+		{
+			gxacts_limit = 1;
```

Comment:
I wonder if we can change the value of `max_tm_gxacts` directly here instead of introducing a new parameter?

Reply:
I also thought of this idea but finally rejected it - it seems unsafe to modify a global variable `max_tm_gxacts` for fault injection purpose (anyway we have to restore it after injecting a fault).